contracts/ronin/validator/SlashingExecution.sol

```diff
@@ -60,10 +59,34 @@ abstract contract SlashingExecution is
 
     // Note: Removing rewards of validator in `bailOut` function is not needed, since the rewards have been
     // removed previously in the `slash` function.
-    _miningRewardBailoutCutOffAtPeriod[_validatorAddr][_period] = true;
-    _miningRewardDeprecatedAtPeriod[_validatorAddr][_period] = false;
-    _blockProducerJailedBlock[_validatorAddr] = block.number - 1;
+    assembly {
+      mstore(0x00, _validatorAddr)
+      mstore(0x20, _miningRewardBailoutCutOffAtPeriod.slot)
+      mstore(0x20, keccak256(0x00, 0x40))
+      mstore(0x00, _period)
+      let key := keccak256(0x00, 0x40)
+      sstore(key, 1)
 
-    emit ValidatorUnjailed(_validatorAddr, _period);
+      mstore(0x00, _validatorAddr)
+      mstore(0x20, _miningRewardDeprecatedAtPeriod.slot)
+      mstore(0x20, keccak256(0x00, 0x40))
+      mstore(0x00, _period)
+      key := keccak256(0x00, 0x40)
+      sstore(key, 0)
+
+      mstore(0x00, _validatorAddr)
+      mstore(0x20, _blockProducerJailedBlock.slot)
+      key := keccak256(0x00, 0x40)
+      sstore(key, sub(number(), 1))
```

Comment:
In general, I don't think writing assembly a whole big part like this is a good idea. Maintainers and auditors will be suffered when reading this.

Reply:
The assembly code above is mainly to guild the compiler to reuse one stack slot (let key) to store the hashing slot for 3 deleting operations